src/main/java/ch/njol/skript/effects/EffToggle.java

```diff
@@ -123,70 +132,80 @@ public boolean init(final Expression<?>[] vars, final int matchedPattern, final
 	
 	@Override
 	protected void execute(final Event e) {
-		if (!flattening) {
-			executeLegacy(e);
-			return;
-		}
-		
-		// 1.13 and newer: use BlockData
-		for (Block b : blocks.getArray(e)) {
-			BlockData data = b.getBlockData();
-			if (toggle == -1) {
-				if (data instanceof Openable)
-					((Openable) data).setOpen(false);
-				else if (data instanceof Powerable)
-					((Powerable) data).setPowered(false);
-			} else if (toggle == 1) {
-				if (data instanceof Openable)
-					((Openable) data).setOpen(true);
-				else if (data instanceof Powerable)
-					((Powerable) data).setPowered(true);
-			} else {
-				if (data instanceof Openable) // open = NOT was open
-					((Openable) data).setOpen(!((Openable) data).isOpen());
-				else if (data instanceof Powerable) // power = NOT power
-					((Powerable) data).setPowered(!((Powerable) data).isPowered());
+		ArrayList<Object> toggledValues = new ArrayList<>();
+		for (Object o : toggledExpr.getArray(e)) {
+			if (o instanceof Block) {
+				Block block = (Block) o;
+				if (!flattening) {
+					executeLegacy(block);
+					return;
+				}
+				
+				// 1.13 and newer: use BlockData
+				BlockData data = block.getBlockData();
+				if (toggle == -1) {
+					if (data instanceof Openable)
+						((Openable) data).setOpen(false);
+					else if (data instanceof Powerable)
+						((Powerable) data).setPowered(false);
+				} else if (toggle == 1) {
+					if (data instanceof Openable)
+						((Openable) data).setOpen(true);
+					else if (data instanceof Powerable)
+						((Powerable) data).setPowered(true);
+				} else {
+					if (data instanceof Openable) // open = NOT was open
+						((Openable) data).setOpen(!((Openable) data).isOpen());
+					else if (data instanceof Powerable) // power = NOT power
+						((Powerable) data).setPowered(!((Powerable) data).isPowered());
+				}
```

Comment:
You can horten this code using this technique
```suggestion
				BlockData data = block.getBlockData();
				if (toggle == 0) {
					if (data instanceof Openable) // open = NOT was open
						((Openable) data).setOpen(!((Openable) data).isOpen());
					else if (data instanceof Powerable) // power = NOT power
						((Powerable) data).setPowered(!((Powerable) data).isPowered());
				} else {
					if (data instanceof Openable)
						((Openable) data).setOpen(toggle == 1);
					else if (data instanceof Powerable)
						((Powerable) data).setPowered(toggle == 1);
				}
```

Reply:
Why was this resolved?